### DISCUSSION TITLE
Call C functions in F'prime

### DISCUSSION THREAD

**Discussion by sheershak-agarwal** [nasa/fprime#842] [Q&A] [answered]
Can I call C function from my components? Will the makefiles compile C code too or I have to edit them to compile C files?

**Comment by LeStarch** [comment, marked as answer]
Yes, you can.  CMake can compile C sources and C++ can call these functions like one would normally call C functions.  Define their prototypes in an `extern "C" {}` block, if I am remembering properly.
